tiling.js

```diff
@@ -3868,15 +3874,35 @@ export function insertWindow(metaWindow, { existing }) {
         resizeHandler(metaWindow);
         connectSizeChanged(true);
 
-        // remove winprop props after window shown
+        // // remove winprop props after window shown
         callbackOnActorShow(actor, () => {
             delete metaWindow.preferredWidth;
 
             Main.activateWindow(metaWindow);
             ensureViewport(space.selectedWindow, space);
             space.setSpaceTopbarElementsVisible(true);
-        });
 
+            switch (Settings.prefs.open_window_position) {
+            case Settings.OpenWindowPositions.DOWN:
+                stackSlurpTimeout = GLib.timeout_add(
+                    GLib.PRIORITY_DEFAULT,
+                    100,
+                    () => {
+                        slurp(active, SlurpInsertPosition.BELOW);
+                        return false; // on return false destroys timeout
+                    });
+                break;
+            case Settings.OpenWindowPositions.UP:
+                stackSlurpTimeout = GLib.timeout_add(
+                    GLib.PRIORITY_DEFAULT,
+                    100,
+                    () => {
+                        slurp(active, SlurpInsertPosition.ABOVE);
+                        return false; // on return false destroys timeout
+                    });
+                break;
+            }
+        });
```

Comment:
Any way to make the animation cleaner? Right now it opens to the side and immediately snaps down as if the keybind was used

Reply:
Not currently - the whole slurping and layouting concept is built on the assumption that windows need to be correctly tiled before slurping.